providers/catalog_test.go

```diff
@@ -93,6 +93,31 @@ var testCases = []struct { // nolint
 		},
 		expectedErr: nil,
 	},
+	{
+		provider:    Zendesk,
+		description: "Zendesk provider config with valid & invalid substitutions",
+		substitutions: map[string]string{
+			"nonexistentvar": "test",
+		},
+		expected: &ProviderInfo{
+			Support: Support{
+				Read:      false,
+				Write:     false,
+				BulkWrite: false,
+				Subscribe: false,
+				Proxy:     true,
+			},
+			AuthType: Oauth2,
+			OauthOpts: OauthOpts{
+				AuthURL:                   "https://d3v-ampersand.zendesk.com/oauth/authorizations/new",
```

Comment:
This AuthURL was tested successfully by following [this documentation](https://ampersand.slab.com/posts/adding-a-new-proxy-connector-mu7822bu#hcjnz-3-update-connectors-providers-catalog-test-go). It would eventually redirect me to the url with the code parameter, but I was unable to test the callback url handing the code and then using the TokenURL.